testing/square_bracket_accessor_test.py

```diff
@@ -0,0 +1,150 @@
+            ['_rowName', "a[ 'foo'   ]"],
+            ['result', 'foo']
+        ])
+
+    def test_string_inner_whitespaces(self):
+        res = mldb.query("""
+            SELECT a['f b'] FROM (
+                SELECT parse_json('{"f b" : 123}') AS a
+            )
+                         """)
+        self.assertEqual(res, [
+            ['_rowName', "a['f b']"],
+            ['result', 'f b']
+        ])
+
+    def test_string_utf8(self):
+        res = mldb.query("""
+            SELECT a['utéf8'] FROM (
+                SELECT parse_json('{"utéf8" : 123}') AS a
+            )
+                         """)
+        self.assertEqual(res, [
+            [u'_rowName', u"a['utéf8']"],
+            [u'result', u'utéf8']
+        ])
+
+    @unittest.expectedFailure
+    def test_match(self):
+        """
+        It doesn't matter that it maches, it's a string.
+        """
+        res = mldb.query("SELECT a[foo] FROM (SELECT {foo: 123} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', "a[foo]"],
+            ['result', 123]
+        ])
+
+    @unittest.expectedFailure
+    def test_no_match(self):
+        res = mldb.query("SELECT a[bar] FROM (SELECT {foo: 123} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', "a[bar]"],
+            ['result', None]
+        ])
+
+    @unittest.expectedFailure
+    def test_chaining(self):
+        res = mldb.query("SELECT a[f][b] FROM (SELECT {f: {b: 123}} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', "a[f][b]"],
+            ['result', 123]
+        ])
+
+    @unittest.expectedFailure
+    def test_mixing(self):
+        res = mldb.query("SELECT a.f[b] FROM (SELECT {f: {b: 123}} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', 'a.f[b]'],
+            ['result', 123]
+        ])
+
+        res = mldb.query("SELECT a[f].b FROM (SELECT {f: {b: 123}} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', 'a.f[b]'],
+            ['result', 123]
+        ])
+
+    @unittest.expectedFailure
+    def test_outter_whitespaces(self):
+        res = mldb.query("SELECT a[ foo   ] FROM (SELECT {foo: 123} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', "a[foo]"],
+            ['result', 123]
+        ])
+
+    @unittest.expectedFailure
+    def test_inner_whitespaces(self):
+        res = mldb.query("""
+            SELECT a[f b] FROM (
+                SELECT parse_json('{"f b" : 123}') AS a
+            )
+                         """)
+        self.assertEqual(res, [
+            ['_rowName', "a[f b]"],
+            ['result', 123]
+        ])
+
+    @unittest.expectedFailure
+    def test_utf8(self):
+        res = mldb.query("""
+            SELECT a[utéf8] FROM (
+                SELECT parse_json('{"utéf8" : 123}') AS a
+            )
+                         """)
+        self.assertEqual(res, [
+            [u'_rowName', u"a[utéf8]"],
+            [u'result', 123]
+        ])
+
```

Comment:
Exactly... they should be the same thing.  However, a["hello world"] is valid, whereas a[hello world] should be a syntax error.